### DISCUSSION TITLE
DatePicker Range pagination only one month

### DISCUSSION THREAD

**Discussion by andresgutgon** [adobe/react-spectrum#4436] [Q&A] [answered]
## What's the problem?
Hi, I'm using React Aria for building our own Date picker and date picker range and it's looking great so far. 

But I'm stack with one problem ⚠️ 

## Moving only one month when clicking next or prev 
For example AirBnB do this 👇 

https://user-images.githubusercontent.com/49499/234338401-b67f9605-7e4f-4e05-a57c-2d2648a25e97.mov

Note how clicking on `Next Month` the months that appear are `May and Jun`. In the first screen were `April and May`

## The code

Looking into the code I don't know how to change the state to make this happen. Any help will be awesome. 

Btw the work you've done on these libraries is amaizing 🙌

**Comment by LFDanLu** [comment]
https://github.com/adobe/react-spectrum/issues/3284 is related, some places in the code to look would be [here](https://github.com/adobe/react-spectrum/blob/1a38c6f7ed998fae8ef5321b832a4d68b7926a5e/packages/%40react-aria/calendar/src/useCalendarBase.ts#L100) and [here](https://github.com/adobe/react-spectrum/blob/1a38c6f7ed998fae8ef5321b832a4d68b7926a5e/packages/%40react-stately/calendar/src/useCalendarState.ts#L210-L220). `visibleDuration` currently matches the number of months visible but to enable the above behavior we'd have to have a separate prop for `pageSize` or have the next/prev buttons always advance by a month/week/some base amount depending on what the calendar is rendering.

**Comment by andresgutgon** [comment]
Hi thanks @LFDanLu for the response! Yes #3284 looks exactly our case. In that issue you said a task was [added to the backlog](https://github.com/adobe/react-spectrum/issues/3284#issuecomment-1204455500) 

Could you share here the task in the backlog to start following?

**Reply by LFDanLu** [reply, marked as answer]
Sure, https://github.com/orgs/adobe/projects/19/views/4?pane=issue&itemId=26599894 is our tracker in the project board, you can also follow https://github.com/adobe/react-spectrum/issues/3284 for any updates as we will link any relevant PRs to that ticket.

**Reply by andresgutgon** [reply]
Great thanks @LFDanLu !
